src/Model/ModelData.php

```diff
@@ -505,87 +430,46 @@ protected function objCacheClear()
      *
      * @return object|DBField|null The specific object representing the field, or null if there is no
      * property, method, or dynamic data available for that field.
-     * Note that if there is a property or method that returns null, a relevant DBField instance will
-     * be returned.
      */
     public function obj(
         string $fieldName,
         array $arguments = [],
-        bool $cache = false,
-        ?string $cacheName = null
+        bool $cache = false
     ): ?object {
-        if ($cacheName !== null) {
-            Deprecation::noticeWithNoReplacment('5.4.0', 'The $cacheName parameter has been deprecated and will be removed');
-        }
-        $hasObj = false;
-        if (!$cacheName && $cache) {
-            $cacheName = $this->objCacheName($fieldName, $arguments);
-        }
-
         // Check pre-cached value
-        $value = $cache ? $this->objCacheGet($cacheName) : null;
-        if ($value !== null) {
-            return $value;
-        }
-
-        // Load value from record
-        if ($this->hasMethod($fieldName)) {
-            $hasObj = true;
-            $value = call_user_func_array([$this, $fieldName], $arguments ?: []);
-        } else {
-            $hasObj = $this->hasField($fieldName) || ($this->hasMethod("get{$fieldName}") && $this->isAccessibleMethod("get{$fieldName}"));
-            $value = $this->$fieldName;
-        }
-
-        // Return null early if there's no backing for this field
-        // i.e. no poperty, no method, etc - it just doesn't exist on this model.
-        if (!$hasObj && $value === null) {
-            return null;
-        }
-
-        // Try to cast object if we have an explicit cast set
-        if (!is_object($value)) {
-            $castingHelper = $this->castingHelper($fieldName, false);
-            if ($castingHelper !== null) {
-                $valueObject = Injector::inst()->create($castingHelper, $fieldName);
-                $valueObject->setValue($value, $this);
-                $value = $valueObject;
+        $value = $cache ? $this->objCacheGet($fieldName, $arguments) : null;
+        if ($value === null) {
+            $hasObj = false;
+            // Load value from record
+            if ($this->hasMethod($fieldName)) {
+                // Try methods first - there's a LOT of logic that assumes this will be checked first.
+                $hasObj = true;
+                $value = call_user_func_array([$this, $fieldName], $arguments ?: []);
+            } else {
+                $getter = "get{$fieldName}";
+                $hasGetter = $this->hasMethod($getter) && $this->isAccessibleMethod($getter);
+                // Try fields and getters if there was no method with that name.
+                $hasObj = $this->hasField($fieldName) || $hasGetter;
```

Comment:
```suggestion
                $hasObj = $hasGetter || $this->hasField($fieldName);
```
slightly more performant if $hasGetter is true

Reply:
Probably doesn't make a noticable difference in performance but I've made the change.